src/main.rs

```diff
@@ -67,16 +73,27 @@ fn subtract<P: AsRef<Path>>(
     scaled: usize,
     hash_function: HashFunctions,
     output: Option<P>,
+    track_abundance: bool,
 ) -> Result<()> {
     info!("Loading queries");
 
     let max_hash = max_hash_for_scaled(scaled as u64);
-    let template_mh = KmerMinHash::builder()
-        .num(0u32)
-        .ksize(ksize as u32)
-        .hash_function(hash_function)
-        .max_hash(max_hash)
-        .build();
+    let template_mh = if track_abundance { 
+        KmerMinHash::builder()
+            .num(0u32)
+            .ksize(ksize as u32)
+            .hash_function(hash_function)
+            .max_hash(max_hash)
+            .abunds(Some(vec![]))
+            .build()
+    } else { 
+        KmerMinHash::builder()
+            .num(0u32)
+            .ksize(ksize as u32)
+            .hash_function(hash_function)
+            .max_hash(max_hash)
+            .build()
+    };
```

Comment:
Another way of writing this block that avoids duplicating the `KmerMinHash::builder()` stuff is

```suggestion
    let abunds = if track_abundance { 
        Some(vec![])
    } else {
        None
    };
    
    let template_mh = KmerMinHash::builder()
        .num(0u32)
        .ksize(ksize as u32)
        .hash_function(hash_function)
        .max_hash(max_hash)
        .abunds(abunds)
        .build();
```

Maintenance-wise it avoids changing the parameters in one block and forgetting to update the other =]